zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterUtils.java

```diff
@@ -24,18 +24,27 @@
 import java.net.InetSocketAddress;
 import java.net.ServerSocket;
 import java.net.Socket;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  *
  */
 public class RemoteInterpreterUtils {
   static Logger LOGGER = LoggerFactory.getLogger(RemoteInterpreterUtils.class);
-  public static int findRandomAvailablePortOnAllLocalInterfaces() throws IOException {
+  private static Set<Integer> portsUsing = new HashSet();
```

Comment:
How do you think, Is there another way to do that, rather than introducing a static state? 
I.e I have hard time imagining how this can be tested..